### PR TITLE
lint: add trailing whitespace and missing newline checks

### LINKED ISSUE #4581: Create a linter for trailing whitespace
we need a linter test for this: there should be no trailing whitespace on any source code.

_Originally posted by @jandubois in https://github.com/lima-vm/lima/pull/4187#discussion_r2784832975_

### PR DESCRIPTION
**Pull request by jandubois**
Remove trailing whitespace from all source files and ensure all text files end with a newline. Add lint checks to prevent reintroduction of both issues.

Fixes #4581

### CODE REVIEW

**Comment by jandubois** [comment]
While fixing one of the test failures I realized that YAML files actually can have semantically significant trailing white space in text blocks. I think Markdown can have it too.

Personally I consider those to be mis-features, as the trailing whitespace is often not visible in the shell, so I would prefer that we don't use these features. But I wanted to point this out, in case somebody disagrees.

File: Makefile [hunk 1]
```diff
@@ -576,6 +576,16 @@ bats: native limactl-plugins
 
 .PHONY: lint
 lint: check-generated
```
Comment (AkihiroSuda) [inline Makefile:578]: Not executed in CI

File: Makefile [hunk 2]
```diff
@@ -576,6 +576,16 @@ bats: native limactl-plugins
...
+	# Check for trailing whitespace. -I skips binary files (images, etc.)
+	! git grep -rn -I '[[:space:]]$$' -- . || \
+		(echo "Please remove trailing whitespace from the lines listed above" && false)
+	# Check that all text files end with a newline. -I skips most binary
+	# files, but not .pb.desc (protobuf descriptors), which contain enough
+	# text-like content to appear as text to git.
+	@result=$$(git grep -rIl '' -- . ':!*.pb.desc' | while IFS= read -r f; do \
+		test "$$(tail -c 1 "$$f")" && echo "$$f: missing newline at end of file"; \
+	done); \
+	if [ -n "$$result" ]; then echo "$$result"; echo "Please ensure all text files end with a newline"; false; fi
```
Comment (AkihiroSuda) [inline Makefile:588]: Maybe we can rather use https://github.com/editorconfig-checker/editorconfig-checker or something else existing

**Comment by AkihiroSuda** [comment]
> from
[add-trailing-whitespace-linter](https://github.com/lima-vm/lima/tree/add-trailing-whitespace-linter)

Next time please use your fork

**Comment by jandubois** [comment]
> Next time please use your fork

Sorry, that was an accident; I normally do.
